framework/src/config/config_reader.rs

```diff
@@ -309,14 +316,17 @@ pub fn read_configuration_from_str(
         cache_size: cache_size,
         ports: ports,
         dpdk_args: None,
+        log_level: log_level,
     })
 }
 
 /// Read a configuration file and create a `NetbricksConfiguration` structure.
 /// `filename` should be TOML formatted file.
 pub fn read_configuration(filename: &str) -> Result<NetbricksConfiguration> {
     let mut toml_str = String::new();
-    let _ = try! {File::open(filename).and_then(|mut f| f.read_to_string(&mut toml_str))
-    .chain_err(|| ErrorKind::ConfigurationError(String::from("Could not read file")))};
+    let _ = try! {
+        File::open(filename).and_then(|mut f| f.read_to_string(&mut toml_str))
+            .map_err(|e| NBError::ConfigurationError(format!("Could not read file due to error: {}", e)))
```

Comment:
Instead of losing the original error here by converting it to string, we can pass that into the `ConfigurationError` as the `cause` maybe.